rotkehlchen/externalapis/hyperliquid.py

```diff
@@ -0,0 +1,128 @@
+import logging
+from collections import defaultdict
+from http import HTTPStatus
+from typing import Any, Literal
+
+import requests
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.assets.utils import symbol_to_asset_or_token
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.fval import FVal
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_fval
+from rotkehlchen.types import ChecksumEvmAddress, Location
+from rotkehlchen.utils.network import create_session
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def asset_from_hyperliquid(symbol: str) -> Asset:
+    """Fetch asset from hperliquid location
+
+    May raise:
+    - UnsupportedAsset
+    - UnknownAsset
+    """
+    return symbol_to_asset_or_token(GlobalDBHandler.get_assetid_from_exchange_name(
+        exchange=Location.HYPERLIQUID,
```

Comment:
So you want to treat it like an exchange with mappings and that's why you added a Location for it? I don't like this. This is breaking our pattern. It looks like a protocol to me and I would not add a location